mbed_host_tests/host_tests_plugins/host_test_plugins.py

```diff
@@ -107,16 +108,41 @@ def print_plugin_char(self, char):
         stdout.flush()
         return True
 
-    def check_mount_point_ready(self, destination_disk, init_delay=0.2, loop_delay=0.25):
+    def check_mount_point_ready(self, destination_disk, init_delay=0.2, loop_delay=0.25, target_id=None):
         """! Waits until destination_disk is ready and can be accessed by e.g. copy commands
-
         @return True if mount point was ready in given time, False otherwise
-
         @param destination_disk Mount point (disk) which will be checked for readiness
         @param init_delay - Initial delay time before first access check
         @param loop_delay - polling delay for access check
         """
+
+        if target_id:
+            # Wait for mount point to appear with mbed-ls
+            # and if it does check if mount point for target_id changed
+            # If mount point changed, use new mount point and check if its ready (os.access)
+            new_destination_disk = destination_disk
+
+            # Sometimes OSes take a long time to mount devices (up to one minute).
+            # Current pooling time: 120x 500ms = 1 minute
+            for i in range(120):
+                # mbed_lstools.create() should be done inside the loop.
+                # Otherwise it will loop on same data.
+                mbeds = mbed_lstools.create()
+                mbeds_by_tid = mbeds.list_mbeds_by_targetid()   # key: target_id, value mbedls_dict()
+                if target_id in mbeds_by_tid:
+                    if 'mount_point' in mbeds_by_tid[target_id]:
+                        new_destination_disk = mbeds_by_tid[target_id]['mount_point']
+                        break
+                self.print_plugin_info("Waiting for '%s' mount point (current is '%s')..."% (target_id, destination_disk))
```

Comment:
I don't think you'll want to print this every half a second. I think you'll want to print it once, with a warning saying that you'll wait for a minute before failing. Then proceed to check every half a second. Otherwise the log is gonna get pretty crazy :)